### PR TITLE
Discrete distributions

### PR DESCRIPTION
**Pull request by maymoo99**
Various discrete distributions other than binomial

### CODE REVIEW

File: stdlib/discreteDistributions.hk [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#	definitions for distributions with discrete outcomes
+# author: Mahmoud Khattab
+
+####################
+# Discrete Uniform #
+####################
+
+# Discrete rectangular distribution.
+# Distribution ranges from 0 to n (inclusive) all with the same probability
+def rectangular(n nat):
```
Comment (zaxtax) [inline stdlib/discreteDistributions.hk]: This needs a better name.

Comment (maymoo99) [reply stdlib/discreteDistributions.hk]: The name is based on [this document](http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Rectangular.pdf) from [this website](http://www.math.wm.edu/~leemis/chart/UDR/UDR.html) which Dr. Carette supplied my capstone group with to base the standard library on.

I don't think it's a good name either but I'm not sure what else to call it. I've tried to looking up an alternate name with no luck

Comment (zaxtax) [reply stdlib/discreteDistributions.hk]: Maybe uniformRange?

Comment (ccshan) [reply stdlib/discreteDistributions.hk]: How about `natUniform`, and rename `discreteUniform` to `intUniform`?

Comment (JacquesCarette) [reply stdlib/discreteDistributions.hk]: Or we could change uniform to take 2 arguments like Lebesgue?

Comment (zaxtax) [reply stdlib/discreteDistributions.hk]: :+1: for natUniform and intUniform. In a perfect world, we would have polymorphism and uniform would have a type like Semiring a => a -> a -> Measure a

**Comment by maymoo99** [comment]
Rebased and resubmitted changes

**Pull request closed without merging**